src/prefect/agent.py

```diff
@@ -66,19 +58,24 @@ def __init__(
             self.default_infrastructure = Process()
             self.default_infrastructure_document_id = None
 
-    async def work_queue_id_from_name(self) -> Optional[UUID]:
+    async def get_work_queues(self) -> Optional[UUID]:
         """
-        For agents that were provided a work_queue_name, rather than a work_queue_id,
-        this function will retrieve the work queue ID corresponding to that name.
-        If no matching queue is found, a warning is logged and `None` is returned.
+        Loads the work queue objects corresponding to the agent's target work queues. If any of them don't exist, they are created.
         """
-        if not self.work_queue_name:
-            raise ValueError("No work queue name provided.")
-        try:
-            work_queue = await self.client.read_work_queue_by_name(self.work_queue_name)
-            return work_queue.id
-        except ObjectNotFound:
-            return await self.client.create_work_queue(name=self.work_queue_name)
+        work_queues = []
+        for name in self.work_queues:
+            try:
+                # support IDs and names
+                if isinstance(name, UUID):
+                    work_queue = await self.client.read_work_queue(id=name)
+                else:
+                    work_queue = await self.client.read_work_queue_by_name(name)
+            except ObjectNotFound:
+                work_queue = await self.client.create_work_queue(
```

Comment:
Might be nice to have a log here

Reply:
We may also need to handle idempotency here in the eventually. What if I scale my agents horizontally and there's a race condition where multiple agents create the same queue? The slower one would crash. Perhaps that's okay with pod restarts.

Reply:
Added a protection against this. Rather than re-entering a read/create try/except pattern, if the create fails we just continue without that work queue. We'll pick it up on the next go-round (and most likely, some other agent is running it anyway)